### ISSUE TITLE
[Improvement] revisit all the dropXXX interfaces to have  a consistent behavior

### ISSUE THREAD

**Issue by FANNG1** [datastrato/gravitino#2434]
### What would you like to be improved?

we should revisit all the dropXXX interfaces:

a) We should define a clear behavior of return value (false) and exception thrown.
b) We should have a consistent behavior for all the dropXXX interfaces.

### Subtasks

- [x] #2913

**Comment by jerryshao** [comment]
@mchades @FANNG1 @yuqi1129 @diqiu50 let's discuss here.

**Comment by FANNG1** [comment]
if succ, return true, if not exists return false, otherwise, throw corresponding exception, the behavior seems clear.

**Comment by jerryshao** [comment]
I think we should revisit all the drop related APIs and implementations, and change the behavior if necessary.

**Comment by charliecheng630** [comment]
I would love to revisit all the drop related APIs and  implementations first, then write the results into a document. 
What do you think about this? @mchades @FANNG1  @yuqi1129  @diqiu50 @jerryshao

**Comment by jerryshao** [comment]
Great, you can go ahead. Thanks a lot.

**Comment by charliecheng630** [comment]
Is this a list of all Rest APIs?
https://github.com/datastrato/gravitino/tree/main/docs/open-api

**Comment by mchades** [comment]
> Is this a list of all Rest APIs? https://github.com/datastrato/gravitino/tree/main/docs/open-api

Yep, you can also look into the codes in package `com.datastrato.gravitino.server.web.rest`

**Comment by charliecheng630** [comment]
@mchades @FANNG1 @yuqi1129 @diqiu50 @jerryshao 
The result after revisiting the drop APIs is following.  Please correct me if there are any omissions.
[Drop Related APIs](https://docs.google.com/spreadsheets/d/1NQX_dZ_KaMmCzC6P60NExyzd37K8_HFnxV9Re057YZs/edit?usp=sharing)

@mchades 
I didn't find the information about `dropPartition` in the open API.   Has it not been made available yet?

**Comment by mchades** [comment]
Could you please also add the exception response? For example, the response when encountering a RuntimeException.

> I didn't find the information about dropPartition in the open API. Has it not been made available yet?

yep, it's WIP #2492 

BTW, it's better to make your doc commentable

**Comment by charliecheng630** [comment]
> Could you please also add the exception response? For example, the response when encountering a RuntimeException.
> 
> > I didn't find the information about dropPartition in the open API. Has it not been made available yet?
> 
> yep, it's WIP #2492
> 
> BTW, it's better to make your doc commentable

The exception response has been updated, but it seems that the implementation is not aligned.

**Comment by jerryshao** [comment]
@charliecheng630 are you still working on this? I found that for most the JDBC catalogs, the current implementation is not aligned the definition, we should improve it.

**Comment by charliecheng630** [comment]
@jerryshao I haven't started to work yet. I will work on this ASAP.

**Comment by jerryshao** [comment]
OK, thanks!

**Comment by charliecheng630** [comment]
@mchades After tracing code, here are the areas that need to be improve：
```
com.datastrato.gravitino.catalog.xxx.xxxCatalogOperations#dropXXX
com.datastrato.gravitino.catalog.xxxOperationDispatcher#dropXXX
com.datastrato.gravitino.storage.relational.service.xxxService#deleteXXX
```
Please correct me if I am wrong.

**Comment by jerryshao** [comment]
I'm working on the core part to fix the current fix, you can check #3075 and the related PR. I think what you should check is:

1. all the specific catalog's dropXXX.
2. Rest server's dropXXX.
3. Client API's dropXXX.

**Comment by jerryshao** [comment]
I've check all the Operation Dispatcher's dropXXX in #3075.
